derive/src/lib.rs

```diff
@@ -256,3 +265,16 @@ fn visit_children(
         Data::Union(_) => unimplemented!(),
     }
 }
+
+fn is_option(ty: &Type) -> bool {
+    if let Type::Path(TypePath { path: Path { segments, .. }, .. }) = ty {
+        if let Some(segment) = segments.last() {
+            if segment.ident == "Option" {
```

Comment:
it seems in theory this would match anything called `Option` (even if it wasn't `std::Option`) but I think that seems ok to me (I don't think we'll have anything else realistically)

Reply:
Agreed, I think we won't have another struct called `Option` in this project 🤔. Or we can match the full path with `std::option::Option` or `core::option::Option` if required.